binderhub/repoproviders.py

```diff
@@ -83,6 +91,138 @@ def get_build_slug(self):
         return '{user}-{repo}'.format(user='Rick', repo='Morty')
 
 
+class GitRepoProvider(RepoProvider):
+    """Bare bones git repo provider.
+
+    Users must provide a spec of the following form.
+
+    <url-escaped-namespace>/<resolved_ref>
```

Comment:
The `GitHubRepoProvider` class doesn't require url escaping right? It just passes it to the tokenizer here:

https://github.com/mariusvniekerk/binderhub/blob/5675b40f9c8082b62a9cb5e3d749d23d28320706/binderhub/repoproviders.py#L28

would this mean that if people are using gitlab Binder, they'd need to escape characters etc, but if using github they don't?

Reply:
I would probably say that we can be smart when generating these spec urls in the application javascript.  Right now we already convert full github urls to the gh/<spec> format.

Reply:
@choldgraf @mariusvniekerk this is for *git*, right? For git, we have to allow arbitrary URLs, so we need to do escaping. GitHub and GitLab both have more specific sturctures that ensure we don't need to do any escaping, since their username/reponame is already URL safe.

I think this should say `url-escaped-url` rather than `url-escaped-namespace`?